kernel/src/process.rs

```diff
@@ -416,46 +416,46 @@ impl Process<'a> {
         }
     }
 
-    pub fn dequeue_task(&mut self) -> Option<Task> {
+    crate fn dequeue_task(&mut self) -> Option<Task> {
         self.tasks.dequeue().map(|cb| {
             self.kernel.decrement_work();
             cb
         })
     }
 
-    pub fn mem_start(&self) -> *const u8 {
+    crate fn mem_start(&self) -> *const u8 {
```

Comment:
I'd suggest keeping this accessors about process memory layout pub. It would be useful, for example, to have a capsule that provides a system call interface allowing a process to find out its layout (e.g., for debugging).

Reply:
The `memop` system call provides this information already, and it's passed into the process's `_start` symbol.

There might be a use for a (probably somewhat privileged) capsule to get this kind of information regardless (e.g. dump the state of the system to a debug log) In any case capsules don't have a way to get a hold of a `Process` struct directly right now, so they couldn't call this method in practice anyway.